proxy/src/control/discovery.rs

```diff
@@ -598,7 +637,94 @@ where T: HttpService<RequestBody = BoxBody, ResponseBody = RecvBody>,
     }
 }
 
-// ===== impl RxError =====
+// ===== impl DstLabels ====
+
+impl DstLabels {
+
+    #[inline]
+    fn new(set_labels: &HashMap<String, String>,
+           addr_labels: &HashMap<String, String>)
+           -> Self
+    {
+        // this could also be a `fold`, or a `map` + `collect`, but the
+        // mutable string + iteration is probably faster.
+        let mut s = String::new();
+        for (k, v) in set_labels.iter().chain(addr_labels.iter()) {
+            write!(s, ",dst_{}=\"{}\"", k, v)
+                .expect("writing to string should not fail");
```

Comment:
how do you know that it's appropriate to prefix the string with `,`? Is it always the case that this string _must_ be prefaced with other labels? that seems extremely subtle. i think, ideally, commas are only appended when joining